### PR TITLE
Clang format final

### PR DESCRIPTION
**Pull request by srajangarg**
Discussions https://github.com/symengine/symengine/pull/882 and https://github.com/symengine/symengine/issues/866
The code coverage test can be ignored, for this PR.

### CODE REVIEW

**Comment by certik** [comment]
This is +1, if tests pass.

**Comment by Sumith1896** [comment]
+1 from me too. 
@isuruf Your views would be great too.

File: bin/travis_clang.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    for FILE in $FILES; do
+
+        $CLANG $FILE | cmp  $FILE >/dev/null
+
+        if [ $? -ne 0 ]; then
+            echo "[!] INCORRECT FORMATTING! $FILE" >&2
+            RETURN=1
+        fi
+
+    done
```
Comment (certik) [inline bin/travis_clang.sh]: This script seems to only check the files changed against master. This seems fragile. Why not to check all the files, always?

Comment (srajangarg) [inline bin/travis_clang.sh]: We anyways have master all formatted up. That's why I thought it was the way to go. Checking all files doesn't hurt. Should I go ahead with thath?

Comment (Sumith1896) [inline bin/travis_clang.sh]: Yes, preferably.

**Comment by certik** [comment]
I think this is good enough, let's merge this. Then we can improve the scripts once it is in master. There might be more things we'll have to iron out.

**Comment by certik** [comment]
@srajangarg I've created #902 and #903 for the remaining issues.

**Comment by certik** [comment]
I am still struggling setting up Vim the right way. I have asked here: http://stackoverflow.com/questions/36364338/

**Pull request merged**